user-carbon/src/utils/Holdings.ts

```diff
@@ -0,0 +1,42 @@
+import { BigInt, BigDecimal, Address } from "@graphprotocol/graph-ts"
+import { CarbonTokenFactory } from "./carbon_token/CarbonTokenFactory"
+import { dayFromTimestamp } from "../../../lib/utils/Dates"
+import { ICarbonToken } from "./carbon_token/ICarbonToken"
+import { Holding } from "../../generated/schema"
+import { KlimateUtils } from "./Klimate"
+
+export class HoldingsUtils {
+
+    private static DAY_IN_SECONDS: BigInt = BigInt.fromString("86400")
+
+    static updateHolding(token: ICarbonToken, timestamp: BigInt, address: Address): void {
+        const dayTimestamp = dayFromTimestamp(timestamp)
+
+        // Handle the to address
+
+        let klimate = KlimateUtils.loadKlimate(address)
+        let holdingId = klimate.id + '-' + token.getToken() + '-' + dayTimestamp
+        let klimateHolding = Holding.load(holdingId)
+        if (klimateHolding == null) {
+            klimateHolding = this.createAndReturnEmptyHolding(holdingId)
+        }
+        let newBalance = token.returnAddressBalance(Address.fromString(klimate.id))
```

Comment:
If this ends up severely slowing indexing in the future, maybe we could use the `Transfer.amount` from event and update current value for a given day. This would require recursion in order to fetch holding for user for previous day when `klimateHolding == null` for the current day.

If this is not a big concern - ignore this comment :)